test/github-issues/8936/issue-8936.ts

```diff
@@ -0,0 +1,56 @@
+import "../../utils/test-setup"
+import {
+    QueryFailedError,
+    QueryRunner,
+    Repository,
+    TableIndex,
+} from "../../../src"
+import {
+    createTestingConnections,
+    closeTestingConnections,
+    reloadTestingDatabases,
+} from "../../utils/test-utils"
+import { Connection } from "../../../src/connection/Connection"
+import { User } from "./entity/User"
+
+describe("github issues > #8936 DropIndex with a TableIndex without name is not working", () => {
+    let connections: Connection[]
+
+    const tableIndex: TableIndex = new TableIndex({
+        columnNames: ["firstName", "lastName"],
+        isUnique: true,
+    })
+
+    before(
+        async () =>
+            (connections = await createTestingConnections({
+                entities: [__dirname + "/entity/*{.js,.ts}"],
+                schemaCreate: true,
+                dropSchema: true,
+            })),
+    )
+    beforeEach(() => reloadTestingDatabases(connections))
+    after(() => closeTestingConnections(connections))
+
+    it("should drop the index as expected", () => {
+        // Create a clone because the createIndex will set the name
+        const dropTableIndex: TableIndex = tableIndex.clone()
+
+        return Promise.all(
+            connections.map(async (connection) => {
+                const queryRunner: QueryRunner = connection.createQueryRunner()
+                const userRepository: Repository<User> =
+                    connection.getRepository(User)
+                const tableName: string = userRepository.metadata.tableName
```

Comment:
I couldn't figure out what the best way was to get the tableName of the entity. If there is a better way I would like to know how so i can change it 😄

Reply:
Also I cannot get the Oracle DB to work locally either on an intel and M1 machine. Is there some extra guidance available?. I checked the Developer documentation already and followed that steps